specs/Workers.md

```diff
@@ -0,0 +1,1399 @@
+  ///
+  /// This corresponds to the `scriptURL` property of the `Worker` object in the DOM.
+  /// The `scriptURL` property returns a string representing the URL of the script that the
+  /// worker is executing. See the [MDN documentation]
+  /// (https://developer.mozilla.org/docs/Web/API/Worker/scriptURL) for more information.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT ScriptUri([out, retval] LPWSTR* value);
+
+  /// Add an event handler for the `Destroyed` event that is raised when the
+  /// worker object is destroyed.
+  /// A worker object is destroyed when the worker script is terminated or when
+  /// the `CoreWebView2Worker` object is destroyed.
+  HRESULT add_Destroyed(
+      [in] ICoreWebView2WorkerDestroyedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Removes an event handler previously added with `add_Destroyed`.
+  HRESULT remove_Destroyed(
+      [in] EventRegistrationToken token);
+}
+
+/// Receives `Destroyed` events.
+[uuid(3197df3b-4245-5fff-9427-e98118fcf657), object, pointer_default(unique)]
+interface ICoreWebView2WorkerDestroyedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Worker* sender,
+      [in] IUnknown* args);
+}
+
+[uuid(f233edb4-d9b4-5209-9abd-b7f50089464c), object, pointer_default(unique)]
+interface ICoreWebView2DedicatedWorker : IUnknown {
+  /// A string specified when creating a dedicated worker. If it is not provided during
+  /// creation, the `Name` property will default to an empty string.
+  ///
+  /// This corresponds to the `options.name` property passed to the `Worker` constructor in the DOM.
+  /// For more information, see the [MDN documentation]
+  /// (https://developer.mozilla.org/docs/Web/API/Worker/Worker).
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Name([out, retval] LPWSTR* value);
+}
+
+
+[uuid(de8ed42b-5128-55df-9dd0-ea1d4d706c87), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorker : IUnknown {
+  /// The state of the service worker. A service worker can be in new, installing,
+  /// installed, activating, activated, or redundant.
+  ///
+  /// This corresponds to the `state` property of the `ServiceWorker` object in the DOM.
+  /// See the [MDN documentation](https://developer.mozilla.org/docs/Web/API/ServiceWorker/state)
+  /// for more information.
+  [propget] HRESULT State([out, retval] COREWEBVIEW2_SERVICE_WORKER_STATE* value);
+}
+
+[uuid(876f8390-f9a8-5264-9677-985d8453aeab), object, pointer_default(unique)]
+interface ICoreWebView2SharedWorker : IUnknown {
+  /// A string specified when creating a shared worker.
```

Comment:
Doesn't mention the default value that the previous Name docs did. Probably want to make sure they are in sync (assuming they should be the same)